File: programs/arrays-arrays_updating-grid-2.java
private int[][] countNeighbors() {
    int rows = grid.numRows();
    int cols = grid.numCols();

    int[][] counts = new int[rows][cols];
    for (int r = 0; r &lt; rows; r++) {
        for (int c = 0; c &lt; cols; c++) {
            counts[r][c] = countAlive(r, c);
        }
    }
    return counts;
}

File: programs/arrays-arrays_updating-grid-2.cpp
private

vector<vector<int>> countNeighbors() {
    int rows = grid->numRows();
    int cols = grid->numCols();

    vector<vector<int>> counts(rows, vector<int>(cols));
    for (int r = 0; r < rows; ++r) {
        for (int c = 0; c < cols; ++c) {
            counts[r][c] = countAlive(r, c);
        }
    }
    return counts;
}
